app/markets/ActivityChart.tsx

```diff
@@ -0,0 +1,192 @@
+  const ratio = total > 0 ? value / total : 0;
+
+  return `${(ratio * 100).toFixed(2)}%`;
+};
+
+interface ActivityChartProps {
+  id: string;
+}
+
+type DataType = { time: string; 0: string; 1: string };
+type BlockDataType = Record<string, { number: string }[]>;
+type OutcomesDataType = Record<string, { outcomeTokenAmounts: string[] }>;
+
+export const ActivityChart = ({ id }: ActivityChartProps) => {
+  const { resolvedTheme } = useTheme();
+
+  const { data: marketData } = useQuery<Pick<Query, 'fixedProductMarketMaker'>>({
+    queryKey: ['getMarket', id],
+  });
+  const market = marketData?.fixedProductMarketMaker;
+
+  const { data: trades } = useQuery({
+    queryKey: ['getMarketTrades', id],
+    queryFn: async () =>
+      getMarketTrades({
+        fpmm: id,
+        orderBy: FpmmTrade_OrderBy.CreationTimestamp,
+      }),
+  });
+
+  const { data } = useQuery({
+    queryKey: ['getMarketOutcomesHistory', id],
+    queryFn: async (): Promise<DataType[] | undefined> => {
+      if (!trades) return;
+
+      const tradesTimestamp = trades.fpmmTrades.map(trade => trade.creationTimestamp);
+
+      const queryBlocks = `query blockNumberByTimestamp {
+          ${tradesTimestamp.map(
+            timestamp => `_${timestamp}: blocks(where: {timestamp: ${timestamp}}, first: 1) {
+              number
+            }`
+          )}
+        }`;
+
+      const blockNumbers = await request<BlockDataType>(
+        XDAI_BLOCKS_SUBGRAPH_URL,
+        queryBlocks
+      );
+
+      let tradesBlockNumbers = Object.values(blockNumbers).map(value => value[0].number);
+      tradesBlockNumbers = Array.from(new Set(tradesBlockNumbers));
+
+      if (!tradesBlockNumbers) return;
+
+      const queryOutcomeTokenAmounts = `query outcomeTokenAmountsByBlockNumber($id: ID!) {
+          ${tradesBlockNumbers.map(
+            timestamp => `_${timestamp}: fixedProductMarketMaker(id: $id, block: { number: ${timestamp} }) {
+              outcomeTokenAmounts
+            }`
+          )}
+        }`;
+
+      const outcomeTokenAmounts = await request<OutcomesDataType>(
+        OMEN_SUBGRAPH_URL,
+        queryOutcomeTokenAmounts,
+        {
+          id,
+        }
+      );
+
+      return Object.values(outcomeTokenAmounts).map((value, index) => {
+        const outcomes = Object.assign({ 0: '', 1: '' }, value.outcomeTokenAmounts);
+        return { time: tradesTimestamp[index], ...outcomes };
+      });
+    },
+    enabled: !!trades,
+  });
+
+  if (!data || !market)
+    return <div className="h-full w-full animate-pulse rounded-8 bg-outline-low-em" />;
+
+  const TooltipContent = (chartData: DefaultTooltipContentProps<number, string>) => {
+    const { payload } = chartData;
+
+    if (!payload || !payload[0] || !payload[1]) return;
+
+    const total = payload.reduce(
+      (result, entry) => (entry.value ? result + +entry.value : 0),
+      0
+    );
+    const tradeTime = format(fromUnixTime(+payload[0].payload.time), 'HH:mm dd/MM/yy');
```

Comment:
should we use a fotmat from `utils/date` or if doesn't existe extracting there?

Reply:
I think this can live here for now.